test/unit/tshelper.spec.ts

```diff
@@ -26,14 +29,132 @@ export class TSHelperTests {
+
+        const decorators = tsHelper.getCustomDecorators(enumType, typeChecker);
+
+        Expect(decorators.size).toBe(1);
+        Expect(decorators.has(DecoratorKind.CompileMembersOnly)).toBeTruthy();
+    }
+
+    @Test("GetCustomDecorators multiple")
+    public GetCustomDecoratorsMultiple(): void {
+        const source = `/** !CompileMembersOnly
+            * !Phantom */
+            enum TestEnum {
+                val1 = 0,
+                val2 = 2,
+                val3,
+                val4 = "bye",
+            }
+
+            const a = TestEnum.val1;`;
+
+        const [sourceFile, typeChecker] = util.parseTypeScript(source);
+        const identifier = util.findFirstChild(sourceFile, ts.isIdentifier);
+        const enumType = typeChecker.getTypeAtLocation(identifier);
+
+        const decorators = tsHelper.getCustomDecorators(enumType, typeChecker);
+
+        Expect(decorators.size).toBe(2);
+        Expect(decorators.has(DecoratorKind.CompileMembersOnly)).toBeTruthy();
+        Expect(decorators.has(DecoratorKind.Phantom)).toBeTruthy();
+    }
+
+    @Test("GetCustomDecorators single jsdoc")
+    public GetCustomDecoratorsSingleJSDoc(): void {
+        const source = `/** @CompileMembersOnly */
+            enum TestEnum {
+                val1 = 0,
+                val2 = 2,
+                val3,
+                val4 = "bye",
+            }
+
+            const a = TestEnum.val1;`;
+
+        const [sourceFile, typeChecker] = util.parseTypeScript(source);
+        const identifier = util.findFirstChild(sourceFile, ts.isIdentifier);
+        const enumType = typeChecker.getTypeAtLocation(identifier);
+
+        const decorators = tsHelper.getCustomDecorators(enumType, typeChecker);
+
+        Expect(decorators.size).toBe(1);
+        Expect(decorators.has(DecoratorKind.CompileMembersOnly)).toBeTruthy();
+    }
+
+    @Test("GetCustomDecorators multiple jsdoc")
+    public GetCustomDecoratorsMultipleJSDoc(): void {
+        const source = `/** @Phantom
+            * @CompileMembersOnly */
+            enum TestEnum {
+                val1 = 0,
+                val2 = 2,
+                val3,
+                val4 = "bye",
+            }
+
+            const a = TestEnum.val1;`;
+
+        const [sourceFile, typeChecker] = util.parseTypeScript(source);
+        const identifier = util.findFirstChild(sourceFile, ts.isIdentifier);
+        const enumType = typeChecker.getTypeAtLocation(identifier);
+
+        const decorators = tsHelper.getCustomDecorators(enumType, typeChecker);
+
+        Expect(decorators.size).toBe(2);
+        Expect(decorators.has(DecoratorKind.Phantom)).toBeTruthy();
+        Expect(decorators.has(DecoratorKind.CompileMembersOnly)).toBeTruthy();
+    }
+
+    @Test("GetCustomDecorators multiple default jsdoc")
+    public GetCustomDecoratorsMultipleDefaultJSDoc(): void {
+        const source = `/**
+            * @description abc
+            * @Phantom
+            * @CompileMembersOnly */
```

Comment:
We should consider using lowercase names for decorators to fit into the jsdoc style.
We could just make them case insensitive in Decorator.ts

Reply:
agreed